src/libraries/System.DirectoryServices.Protocols/ref/System.DirectoryServices.Protocols.cs

```diff
@@ -469,7 +469,9 @@ public partial class PermissiveModifyControl : System.DirectoryServices.Protocol
     public partial class QuotaControl : System.DirectoryServices.Protocols.DirectoryControl
     {
         public QuotaControl() : base (default(string), default(byte[]), default(bool), default(bool)) { }
+        [System.Runtime.Versioning.SupportedOSPlatformAttribute("windows")]
```

Comment:
Should the whole `QuotaControl` type be marked as `SupportedOSPlatformAttribute("windows")`?

https://docs.microsoft.com/en-us/dotnet/api/system.directoryservices.protocols.quotacontrol says

> The QuotaControl class is used to pass the security identifier (SID) of a security principle to retrieve quota-related data.

Would someone ever use this class outside of Windows?

cc @joperezr

Reply:
It is true that today this control is Windows specific, but I'm not sure I'm aware of the implications of adding this attribute to the whole class so I'm not sure how to move forward. What does the actual attribute do? is it just for documentation purposes or does the compiler look at it?

Reply:
The compiler does look at it by way of the Platform Compat Analyzer, which is enabled by default in .NET 5 projects.  Annotating the class would give a warning to any reference of the type in a call site reachable on non-Windows platforms.

Reply:
I see, in that case I think it would make sense to put the attribute in the whole class, since when using this control you really have to specify a SID, so we do want to keep it windows specific for now.

Reply:
Will move the attribute to the type, thanks!